### PR TITLE
Enhance pagination to always show the last page

### PR DESCRIPTION
**Pull request by oliverlloyd**
This one goes out to @paperboyo 

### What
Always show the last page in the pagination control

Screenshot 2020-02-27 at 13 55 01

### Why?
We already have in context the total number of pages so there's no extra cost to doing this and it helps the reader manage their expectations when navigating

### Why not?
There's an argument to say giving an indication of how many pages lie ahead could put people off?

### CODE REVIEW

File: src/components/Pagination/Pagination.tsx [hunk 1]
```diff
@@ -201,24 +203,28 @@ export const Pagination = ({
...
+          />
+        )}
+        {showForthPage && (
+          <PageButton
+            page={forthPage}
+            setPage={setPage}
+            selected={page === forthPage}
+          />
+        )}
+        {showSecondElipsis && <div className={elipsisStyles}>...</div>}
```
Comment (gtrufitt) [inline src/components/Pagination/Pagination.tsx]: `&hellip;`?

File: src/components/Pagination/Pagination.tsx [hunk 2]
```diff
@@ -178,7 +178,9 @@ export const Pagination = ({
   const secondPage = decideSecondPage({ page, pages });
   const thirdPage = decideThirdPage({ page, pages });
   const forthPage = decideForthPage({ page, pages });
-  const showLastPage = page === pages - 2;
+  const showThirdPage = pages > 2;
```
Comment (gtrufitt) [inline src/components/Pagination/Pagination.tsx]: Took me a second to reason with `pages` being `numberOfPages` or `totalPages`, maybe it could be slightly more descriptive?

File: src/components/Pagination/Pagination.tsx [hunk 3]
```diff
@@ -178,7 +178,9 @@ export const Pagination = ({
...
+  const showForthPage = pages > 3;
+  const showLastPage = page < pages - 1;
```
Comment (gtrufitt) [inline src/components/Pagination/Pagination.tsx]: same with `page`, possibly `currentPage`?

File: src/components/Pagination/Pagination.tsx [hunk 4]
```diff
@@ -201,24 +203,28 @@ export const Pagination = ({
...
+            page={thirdPage}
+            setPage={setPage}
+            selected={page === thirdPage}
+          />
+        )}
+        {showForthPage && (
+          <PageButton
+            page={forthPage}
+            setPage={setPage}
+            selected={page === forthPage}
```
Comment (gtrufitt) [inline src/components/Pagination/Pagination.tsx]: `isSelected`?

**Comment by paperboyo** [comment]
Thanks bunch! Not only me too. Check out Pagination (deep-linking doesn't work for some reason: https://www.theguardian.design/2a1e5182b/p/300696-)

File: src/components/Pagination/Pagination.tsx (hunk 3 above)
Comment (oliverlloyd) [reply src/components/Pagination/Pagination.tsx]: Why is my comment not loading? Github?

Comment (oliverlloyd) [reply src/components/Pagination/Pagination.tsx]: I implemented all the other naming changes and it really helps, the code is a lot easier but when I put this one in things got a bit wordy:

```
        {showBackButton && <Back selectedPage={selectedPage} setSelectedPage={setSelectedPage} />}
        <PageButton selectedPage={1} setSelectedPage={setSelectedPage} isSelected={selectedPage === 1} />
        {showFirstElipsis && <div className={elipsisStyles}>...</div>}
        <PageButton
          selectedPage={secondPage}
          setSelectedPage={setSelectedPage}
          isSelected={selectedPage === secondPage}
        />
```

Same but using 'current'

```
        {showBackButton && <Back currentPage={currentPage} setCurrentPage={setCurrentPage} />}
        <PageButton currentPage={1} setCurrentPage={setCurrentPage} isSelected={currentPage === 1} />
        {showFirstElipsis && <div className={elipsisStyles}>...</div>}
        <PageButton
          currentPage={secondPage}
          setCurrentPage={setCurrentPage}
          isSelected={currentPage === secondPage}
        />
```

Or with no new changes:

```
        {showBackButton && <Back page={page} setPage={setPage} />}
        <PageButton page={1} setPage={setPage} isSelected={page === 1} />
        {showFirstElipsis && <div className={elipsisStyles}>...</div>}
        <PageButton
          page={secondPage}
          setPage={setPage}
          isSelected={page === secondPage}
        />
```

What do you think? 🤷‍♂

Comment (gtrufitt) [reply src/components/Pagination/Pagination.tsx]: I like `currentPage` and `isSelected` personally, makes it really easy to comprehend

**Comment by oliverlloyd** [comment]
@gtrufitt Great suggestions! It's so much harder thinking of names when you're buried in the code
